catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/query/QueryApplication.java

```diff
@@ -150,9 +156,32 @@ public void init() {
   private CqlQueryResponse executeCqlQuery(CqlRequest cqlRequest)
       throws UnsupportedQueryException, SourceUnavailableException, FederationException {
     QueryRequest request = cqlRequest.createQueryRequest(catalogFramework.getId(), filterBuilder);
-
     Stopwatch stopwatch = Stopwatch.createStarted();
-    QueryResponse response = catalogFramework.query(request);
+
+    List<QueryResponse> responses = Collections.synchronizedList(new ArrayList<>());
+    List<Result> results =
+        ResultIterable.resultIterable(
+                (queryRequest) -> {
+                  QueryResponse queryResponse = catalogFramework.query(queryRequest);
+                  responses.add(queryResponse);
+                  return queryResponse;
+                },
+                request,
+                cqlRequest.getCount())
+            .stream()
+            .collect(Collectors.toList());
+    QueryResponse response =
+        new QueryResponseImpl(
+            request,
+            results,
+            true,
+            responses.stream().map(QueryResponse::getHits).findFirst().orElse(-1l),
+            responses
+                .stream()
+                .map(QueryResponse::getProperties)
+                .findFirst()
+                .orElse(Collections.emptyMap()));
```

Comment:
The responses collection will always have at least one responses (even if there were 0 hits).
Also, the Response will always have properties. If those conditions hold, line 179-183 collapses to:

`responses.get(0).getProperties()`

Reply:
see https://github.com/codice/ddf/pull/2787#discussion_r170010295